graph-gateway/src/network/internal.rs

```diff
@@ -40,47 +41,25 @@ pub async fn fetch_update(
     state: &InternalState,
 ) -> anyhow::Result<NetworkTopologySnapshot> {
     // Fetch and process the network topology information
-    let (indexers_info, (subgraphs_info, deployments_info)) = futures::future::try_join(
-        async {
-            let info = match tokio::time::timeout(
-                NETWORK_TOPOLOGY_FETCH_TIMEOUT,
-                fetch_and_pre_process_indexers_info(client),
-            )
-            .await
-            {
-                // If the fetch timed out, return an error
-                Err(_) => Err(anyhow!("indexers info fetch timed out")),
-                Ok(resp) => match resp {
-                    // If the fetch failed, return an error
-                    Err(err) => Err(anyhow!("indexers info fetch failed: {err}")),
-                    Ok(resp) => Ok(resp),
-                },
-            }?;
-
-            // Process the fetched network topology information
-            Ok::<_, anyhow::Error>(indexer_processing::process_info(state, info).await)
+    let (subgraphs_info, deployments_info, indexers_info) = match tokio::time::timeout(
+        NETWORK_TOPOLOGY_FETCH_TIMEOUT,
+        fetch_and_pre_process_subgraph_info(client),
+    )
+    .await
+    {
+        // If the fetch timed out, return an error
+        Err(_) => Err(anyhow!("info fetch timed out")),
+        Ok(resp) => match resp {
+            // If the fetch failed, return an error
+            Err(err) => Err(anyhow!(" info fetch failed: {err}")),
```

Comment:
```suggestion
            Err(err) => Err(anyhow!("info fetch failed: {err}")),
```

Reply:
Addressed in [15d3252](https://github.com/edgeandnode/gateway/pull/785/commits/15d3252ecedab95bf24ad9d820d0eac648d18305)